cardano-api/src/Cardano/Api/TxMetadata.hs

```diff
@@ -106,21 +106,18 @@ instance HasTypeProxy TxMetadata where
 
 instance SerialiseAsCBOR TxMetadata where
     serialiseToCBOR =
-          CBOR.serialize'
-        . (Shelley.Metadata :: Map Word64 Shelley.Metadatum -> Shelley.Metadata ())
-        -- The Shelley (Metadata era) is always polymorphic in era,
-        -- so we pick the unit type.
+          -- This is a workaround. There is a tiny chance that serialization could change
+          -- for Metadata in the future, depending on the era it is being used in. For now
+          -- we can pretend like it is the same for all eras starting with Shelley
```

Comment:
How would we replace this work-around in the future?

Reply:
Cool.  Can you add that to the comment?

Reply:
Sure. Will do.

Reply:
Done